modules/apps/data-engine/data-engine-rest-test/src/testIntegration/java/com/liferay/data/engine/rest/resource/v2_0/test/DataDefinitionResourceTest.java

```diff
@@ -769,6 +804,90 @@ private DataDefinition _createDataDefinition(
 		return dataDefinition;
 	}
 
+	private void _testCopiedDataDefinitionDataLayout(
+		DataLayout dataLayout1, DataLayout dataLayout2) {
+
+		String[] dataLayoutColumnFieldNames1 = {};
+
+		for (DataLayoutPage dataLayoutPage : dataLayout1.getDataLayoutPages()) {
+			for (DataLayoutRow dataLayoutRow :
+					dataLayoutPage.getDataLayoutRows()) {
+
+				for (DataLayoutColumn dataLayoutColumn :
+						dataLayoutRow.getDataLayoutColumns()) {
+
+					dataLayoutColumnFieldNames1 =
+						dataLayoutColumn.getFieldNames();
+				}
+			}
+		}
+
+		String[] dataLayoutColumnFieldNames2 = {};
+
+		for (DataLayoutPage dataLayoutPage : dataLayout2.getDataLayoutPages()) {
+			for (DataLayoutRow dataLayoutRow :
+					dataLayoutPage.getDataLayoutRows()) {
+
+				for (DataLayoutColumn dataLayoutColumn :
+						dataLayoutRow.getDataLayoutColumns()) {
+
+					dataLayoutColumnFieldNames2 =
+						dataLayoutColumn.getFieldNames();
+				}
+			}
+		}
```

Comment:
@devpedrotavares please create a private method to reuse code, something like:

```
private List<String> _getDataLayoutColumnFieldNames(DataLayout dataLayout) {
	List<String> dataLayoutColumnFieldNames = new ArrayList<>();

	for (DataLayoutPage dataLayoutPage : dataLayout.getDataLayoutPages()) {
		for (DataLayoutRow dataLayoutRow : dataLayoutPage.getDataLayoutRows()) {
			for (DataLayoutColumn dataLayoutColumn : dataLayoutRow.getDataLayoutColumns()) {
				dataLayoutColumnFieldNames.add(dataLayoutColumn.getFieldNames());
			}
		}
	}

	return dataLayoutColumnFieldNames;
}

private void _testCopiedDataDefinitionDataLayout(DataLayout dataLayout1, DataLayout dataLayout2) {
	List<String> dataLayoutColumnFieldNames1 = _getDataLayoutColumnFieldNames(dataLayout1);
	List<String> dataLayoutColumnFieldNames2  = _getDataLayoutColumnFieldNames(dataLayout2);

	Assert.assertEquals(dataLayoutColumnFieldNames1.size(), dataLayoutColumnFieldNames2.size());

	for (String dataLayoutColumnFieldName : dataLayoutColumnFieldNames1) {
		Assert.assertEquals("CopyOf" + dataLayoutColumnFieldName, dataLayoutColumnFieldNames2.get(i));
	}
}
```

Reply:
Done!